hidimstat/desparsified_lasso.py

```diff
@@ -85,68 +139,246 @@ def desparsified_lasso_confint(X, y, confidence=0.95, max_iter=5000,
-    clf_lasso_cv.fit(X, y)
-    beta_lasso = clf_lasso_cv.coef_
+    sigma_hat, beta_lasso = reid(X, y, n_jobs=n_jobs)
+
+    # Computing the degrees of freedom adjustement
+    if dof_ajdustement:
+        coef_max = np.max(np.abs(beta_lasso))
+        support = np.sum(np.abs(beta_lasso) > 0.01 * coef_max)
+        support = min(support, n_samples - 1)
+        dof_factor = n_samples / (n_samples - support)
+    else:
+        dof_factor = 1
 
-    # Estimating the coefficient vector
-    beta_bias = y.T.dot(Z) / np.sum(X * Z, axis=0)
+    # Computing Desparsified Lasso estimator and confidence intervals
+    beta_bias = dof_factor * np.dot(y.T, Z) / np.sum(X * Z, axis=0)
 
     P = ((Z.T.dot(X)).T / np.sum(X * Z, axis=0)).T
     P_nodiag = P - np.diag(np.diag(P))
+    Id = np.identity(n_features)
+    P_nodiag = dof_factor * P_nodiag + (dof_factor - 1) * Id
 
     beta_hat = beta_bias - P_nodiag.dot(beta_lasso)
 
-    sigma_hat, _ = reid(X, y)
+    omega_diag = omega_diag * dof_factor ** 2
+    omega_invsqrt_diag = omega_diag ** (-0.5)
+
+    quantile = st.norm.ppf(1 - (1 - confidence) / 2)
 
     confint_radius = np.abs(quantile * sigma_hat /
                             (np.sqrt(n_samples) * omega_invsqrt_diag))
     cb_max = beta_hat + confint_radius
     cb_min = beta_hat - confint_radius
 
     return beta_hat, cb_min, cb_max
+
+
+def desparsified_group_lasso(X, Y, cov=None, test='chi2', normalize=True,
+                             max_iter=5000, tol=1e-3, residual_method='lasso',
+                             c=0.01, noise_method='AR', order=1,
+                             n_jobs=1, memory=None, verbose=0):
+    """Desparsified Group Lasso
+
+    Parameters
+    -----------
+    X : ndarray, shape (n_samples, n_features)
+        Data.
+
+    Y : ndarray, shape (n_samples, n_times)
+        Target.
+
+    cov : ndarray, shape (n_times, n_times), optional (default=None)
+        If None, a temporal covariance matrix of the noise is estimated.
+        Otherwise, `cov` is the temporal covariance matrix of the noise.
+
+    test : string, optional (default='chi2')
+        Statistical test used to compute p-values. 'chi2' corresponds
+        to a chi-squared test and 'F' corresponds to an F-test.
+
+    normalize : bool, optional (default=True)
+        If True, the regressors X will be normalized before regression and
+        the target Y will be centered substracting the mean of all the
+        elements of Y to every element of Y.
+
+    max_iter : int, optional (default=5000)
+        The maximum number of iterations when regressing, by Lasso,
+        each column of the design matrix against the others.
+
+    tol : float, optional (default=1e-3)
+        The tolerance for the optimization of the Lasso problems: if the
+        updates are smaller than `tol`, the optimization code checks the
+        dual gap for optimality and continues until it is smaller than `tol`.
+
+    residual_method : string, optional (default='lasso')
```

Comment:
```suggestion
    residual_method : str, optional (default='lasso')
```
the type name in python is str not string